ui/src/testing/stubs/profile-api-stub.ts

```diff
@@ -111,4 +111,9 @@ export class ProfileApiStub extends ProfileApi {
       resolve({});
     });
   }
+
+  public submitDataUseAgreement(dataUseAgreementVersion: number) {
+    return Promise.resolve(this.profile);
```

Comment:
For consistency with the other stubs which actually modify the profile object, maybe you can add a line:

this.profile.dataUseAgreementSignedVersion = dataUseAgreementVersion;